### PR TITLE
sema: make error note correct and check if variable is extern

### LINKED ISSUE #15530: Incorrect error notes when variable is extern
### Zig Version

zig-0.11.0-dev.2892+fd6200eda

### Steps to Reproduce and Observed Behavior

```zig
const x = extern struct {
    y: u4 = 0,
};
```

```
 note: only integers with power of two bits are extern compatible
```
And also

```zig

extern const x: u3;

pub fn main() void {
    _ = x;
}
```
compiles fine.

### Expected Behavior

correct handling of "extern"

### PR DESCRIPTION
**Pull request by wrongnull**
Closes #15530

### CODE REVIEW

**Comment by Vexu** [comment]
> +34,004 −33,990

Did you change whitespace?

**Comment by wrongnull** [comment]
> > +34,004 −33,990
> 
> Did you change whitespace?

 No, I initially mistyped in several places and did not take into account the formatting. Now everything should be fine

**Comment by mlugg** [comment]
No, it's still got the same diff. Almost certainly you've configured your editor to convert whitespace to tabs.
image

**Comment by wrongnull** [comment]
> No, it's still got the same diff. Almost certainly you've configured your editor to convert whitespace to tabs or similar. image

I double-checked it. There are whitespaces instead of tabs.

**Comment by mlugg** [comment]
It might be line endings then - your editor could be configured to convert LF to CRLF.

File: src/Sema.zig [hunk 1]
```diff
@@ -22749,10 +22749,24 @@ fn validateVarType(
...
     is_extern: bool,
 ) CompileError!void {
-    if (try sema.validateRunTimeType(var_ty, is_extern)) return;
 
     const mod = sema.mod;
 
+    if (is_extern) {
+        if (!try sema.validateExternType(var_ty, .other)) {
+            const msg = msg: {
+                const msg = try sema.errMsg(block, src, "variable of type '{}' cannot be extern.", .{var_ty.fmt(mod)});
```
Comment (Vexu) [inline src/Sema.zig:22758]:
```suggestion
                const msg = try sema.errMsg(block, src, "extern variable cannot have type '{}'", .{var_ty.fmt(mod)});
```

**Review by Vexu — APPROVED** [review]
Please add a test for the new error.

**Pull request merged**